### PR TITLE
Feature/remove json dependency

### LINKED ISSUE #32: Explicit 'json' dependency causes build failure with 2.4-head
Hey Linode team, et. al.,

I wouldn't normally report an issue with a pre-release version of Ruby, but the JSON gem problem affects every version of Ruby since 1.9.x, it just manifests in weirder, more subtle ways.  I've been hacking around it since the 2.0 time period.  My production stack is 2.3 stable (vs. an unreleased 2.4), and the 'json' dependency _works_ there, but still creates some quirky situations.

The version of 'json' that has shipped with Ruby since 1.9 is a vendored copy of flori/json, and has since diverged enough with compatibility changes that it looks like a fork.  The gem on Rubygems is the (still-maintained) work in flori/json.

The end result of an explicit dependency on "json" is:
- Assuming Ruby 1.9+, "require 'json'" can mean odd, subtly different things, esp. when the author expected the behavior (or Object even!) from a stock Ruby install.
- Because the builtin 'json' is compiled _with_ Ruby, systems that otherwise wouldn't need build tools (C compiler, ruby-dev, etc) to use the linode gem now do.
- There will be times, like now, where JSON functionality would've worked out of the box, but is broken with a slightly lagging 'json' gem.  MRI's JSON stays in lock-step with MRI.

I don't know what the linode gem's Ruby version policy is, which could be the constraining factor (1.8 still?), but I believe @mperham [summed up the solution that has the best chance of working at the most places](http://www.mikeperham.com/2016/02/09/kill-your-dependencies/):

> So many gems declare a dependency on json, oj, multi_json, or yajl-ruby. There’s so many ossified layers of cruft around JSON processing that only one course of action makes sense: remove it all. JSON has been in the stdlib since 1.9, you don’t need to declare any dependencies at all. Just require 'json' and let Ruby deal with it.
> 
> Rails did it, so can you!

The good thing is: `linode` appears to work just fine with the dependency removed, and all tests pass.  This would've been a one-line PR instead of a novel, but the issues are subtle (more so than "let's break out-of-the-box behavior with 1.8.7!") and I wanted to get some feedback about it.

Thanks

### PR DESCRIPTION
**Pull request by RobertDeRose**
- Removed explicit dependency on JSON gem
- Removed Gemfile.lock as this is a library
- Made Gemfile rely on gemspec to remove duplication
- Updated README to include how to use with older Rubies
- Updated README to encourage using Gemfile and Bundler for Installation
- Updated README to include a CONTRIBUTE section

Close #32

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,12 +1,6 @@
 require 'rubygems'
 require 'rake'
 require 'rake/testtask'
-
-begin
```
Comment (RobertDeRose) [inline Rakefile]: There are no rdocs, so there is no reason to include this task.

File: linode.gemspec [hunk 2]
```diff
@@ -2,8 +2,7 @@
 
 Gem::Specification.new do |s|
   s.name = "linode"
-  s.version = "0.9.0"
-  s.date = "2016-01-02"
+  s.version = File.open("VERSION", "r").read.strip
```
Comment (RobertDeRose) [inline linode.gemspec]: No reason to have to update this and the VERSION file every time you make a version bump, just read the file.

File: README.md [hunk 3]
```diff
@@ -21,7 +74,8 @@ You will need to get an API key (check your account profile).
 Here is an annoyingly exhaustive IRB session where I play around with the API:
 
 ```ruby
-irb> require 'rubygems'
+bundle exec irb
```
Comment (RobertDeRose) [inline README.md]: encourage the use of bundler, it really does make life easier :)

**Comment by RobertDeRose** [comment]
ping @displague

**Comment by displague** [comment]
Looks good to me - I'll either change that changelog post merge/tag or perhaps you can weigh in on the other open pull request, I'll merge that, and you can update the changelog here with all of the changes (mutate/kvmify/fetch_api_key).

**Pull request merged**